src/core/providers/ogr/qgsogrproviderutils.cpp

```diff
@@ -2337,15 +2337,14 @@ void QgsOgrProviderUtils::release( QgsOgrLayer *&layer )
 }
 
 
-void QgsOgrProviderUtils::releaseDataset( QgsOgrDataset *&ds )
+void QgsOgrProviderUtils::releaseDataset( QgsOgrDataset *ds )
 {
   if ( !ds )
     return;
 
   QMutexLocker locker( sGlobalMutex() );
   releaseInternal( ds->mIdent, ds->mDs, true );
   delete ds;
-  ds = nullptr;
```

Comment:
This is not quite the same meaning. Are you sure it won't lead to use after free errors?

Reply:
I don't think so. Same point of view as @m-kuhn : https://github.com/qgis/QGIS/pull/46409#issuecomment-988680806

BTW, we have not experienced any crash with this patch on FreeBSD.